lib/Net/Async/Blockchain/Omni.pm

```diff
@@ -125,23 +144,63 @@ async sub transform_transaction {
 
     return undef unless $transaction_type;
 
-    my $transaction = Net::Async::Blockchain::Transaction->new(
-        currency     => $self->currency_symbol,
-        hash         => $decoded_raw_transaction->{txid},
-        block        => $block,
-        from         => $from->{address},
-        to           => $to->{address},
-        amount       => $amount,
-        fee          => $fee,
-        fee_currency => FEE_CURRENCY,
-        type         => $transaction_type,
-        property_id  => $received_transaction->{propertyid},
-        timestamp    => $received_transaction->{blocktime},
-    );
+    if ($omni_transaction->{type} eq "Send All") {
+
+        for my $data ($omni_transaction->{subsends}->@*) {
+            $sendall{$data->{propertyid}} = $data->{amount};
+        }
+
+        for my $propertyid (keys %sendall) {
+
+             push @transaction , Net::Async::Blockchain::Transaction->new(
+
+                currency     => $self->currency_symbol,
+                hash         => $omni_transaction->{txid},
+                block        => $block,
+                from         => $from->{address},
+                to           => $to->{address},
+                amount       => Math::BigFloat->new($sendall{$propertyid}),
+                fee          => $fee,
+                fee_currency => FEE_CURRENCY,
+                type         => $transaction_type,
+                property_id  => $propertyid,
+                timestamp    => $omni_transaction->{blocktime},
+            );
+        }
+    }
 
-    $self->source->emit($transaction) if $transaction;
+    else {
+
+        @transaction = Net::Async::Blockchain::Transaction->new(
+
+            currency     => $self->currency_symbol,
+            hash         => $omni_transaction->{txid},
+            block        => $block,
+            from         => $from->{address},
+            to           => $to->{address},
+            amount       => $amount,
+            fee          => $fee,
+            fee_currency => FEE_CURRENCY,
+            type         => $transaction_type,
+            property_id  => $omni_transaction->{propertyid},
+            timestamp    => $omni_transaction->{blocktime},
+        );
+    }
 
-    return 1;
+    return @transaction if @transaction;
+
```

Comment:
```suggestion
return undef;
```

Reply:
@chylli-binary good catch.  but need we need to return empty list like `return ();` or `return;`. in case `return undef` we'll have array with 1 element here `my @transaction = await _process_transaction($self, $omni_transaction, $parent_transaction);`